src/librustc_mir/borrow_check/nll/constraint_generation.rs

```diff
@@ -75,14 +84,22 @@ impl<'cx, 'gcx, 'tcx> ConstraintGeneration<'cx, 'gcx, 'tcx> {
                     }
                 });
 
-            self.liveness
-                .drop
-                .simulate_block(self.mir, bb, |location, live_locals| {
+            self.liveness.drop.simulate_block(
+                self.mir,
+                bb,
+                |location, live_locals| {
                     for live_local in live_locals.iter() {
-                        let live_local_ty = self.mir.local_decls[live_local].ty;
-                        self.add_drop_live_constraint(live_local_ty, location);
+                        if let LookupResult::Exact(mpi) =
+                            self.move_data.rev_lookup.find(&Lvalue::Local(live_local))
+                        {
+                            if self.flow_inits.has_any_child_of(mpi).is_some() {
```

Comment:
👍 IFIUK, this is basically the "simple" version, in which we consider the entire variable droppable if any part of it is droppable. Seems good as a starting point.